pkg/registry/core/service/storage/rest_test.go

```diff
@@ -2037,39 +1394,27 @@ func TestServiceRegistryIPUpdate(t *testing.T) {
 	update.Spec.ClusterIP = testIP
 	update.Spec.ClusterIPs[0] = testIP
 
-	_, _, err := storage.Update(ctx, update.Name, rest.DefaultUpdatedObjectInfo(update), rest.ValidateAllObjectFunc, rest.ValidateAllObjectUpdateFunc, false, &metav1.UpdateOptions{})
+	_, _, err = storage.Update(ctx, update.Name, rest.DefaultUpdatedObjectInfo(update), rest.ValidateAllObjectFunc, rest.ValidateAllObjectUpdateFunc, false, &metav1.UpdateOptions{})
 	if err == nil || !errors.IsInvalid(err) {
 		t.Errorf("Unexpected error type: %v", err)
 	}
 }
 
 func TestServiceRegistryIPLoadBalancer(t *testing.T) {
-	storage, registry, server := NewTestREST(t, nil, singleStackIPv4)
+	storage, server := NewTestREST(t, singleStackIPv4)
 	defer server.Terminate(t)
 
-	svc := &api.Service{
-		ObjectMeta: metav1.ObjectMeta{Name: "foo", ResourceVersion: "1"},
-		Spec: api.ServiceSpec{
-			Selector:        map[string]string{"bar": "baz"},
-			SessionAffinity: api.ServiceAffinityNone,
-			Type:            api.ServiceTypeLoadBalancer,
-			Ports: []api.ServicePort{{
-				Port:       6502,
-				Protocol:   api.ProtocolTCP,
-				TargetPort: intstr.FromInt(6502),
-			}},
-		},
-	}
+	svc := svctest.MakeService("foo", svctest.SetTypeLoadBalancer)
 	ctx := genericapirequest.NewDefaultContext()
 	createdSvc, err := storage.Create(ctx, svc, rest.ValidateAllObjectFunc, &metav1.CreateOptions{})
 	if createdSvc == nil || err != nil {
 		t.Errorf("Unexpected failure creating service %v", err)
 	}
-	defer releaseServiceNodePorts(t, ctx, svc.Name, storage, registry)
+	defer releaseServiceNodePorts(t, ctx, svc.Name, storage)
 
 	createdService := createdSvc.(*api.Service)
-	if createdService.Spec.Ports[0].Port != 6502 {
-		t.Errorf("Expected port 6502, but got %v", createdService.Spec.Ports[0].Port)
+	if createdService.Spec.Ports[0].Port != svc.Spec.Ports[0].Port {
```

Comment:
same -- let us refocus tests each covers one aspect of services. or if we have to validate ports in every test then let us create a helper then call it from every test.

Reply:
Let's do that as we move - I'm just trying to clear the way for the delayering.  Not Worse Than Before is the goal :)